pykafka/protocol.py

```diff
@@ -172,10 +172,13 @@ def __init__(self,
         self.produce_attempt = produce_attempt
 
     def __len__(self):
-        size = 4 + 1 + 1 + 4 + 4 + len(self.value)
-        if self.partition_key is not None:
-            size += len(self.partition_key)
-        return size
+        if not self.value==None:
+            size = 4 + 1 + 1 + 4 + 4 + len(self.value)
+            if self.partition_key is not None:
+                size += len(self.partition_key)
+            return size
+        else:
+            return 0
```

Comment:
It could be problematic that this returns the wrong length: if `value` is `None`, there's still the length of the headers plus the length of `partition_key` (it wouldn't be very useful to send a null payload without sending a key, so we should expect `partition_key` to be populated).